sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala

```diff
@@ -169,6 +169,11 @@ case class Literal protected (value: Any, dataType: DataType)
 
   override def toString: String = if (value != null) value.toString else "null"
 
+  override def hashCode(): Int = {
```

Comment:
I think the test failure is due to this patch, and it's an NPE somewhere. It could be because a field you're using in a hashCode() is null, and it seems like can be the case here. Instead of using `.hashCode()`, use `Objects.hashCode` which handles null.